astroid/brain/brain_builtin_inference.py

```diff
@@ -570,12 +574,16 @@ def infer_property(node, context=None):
     prop_func = objects.Property(
         function=inferred,
         name=inferred.name,
-        doc=getattr(inferred, "doc", None),
+        doc=getattr(inferred, "doc_node.value", None),
         lineno=node.lineno,
         parent=node,
         col_offset=node.col_offset,
     )
-    prop_func.postinit(body=[], args=inferred.args)
+    prop_func.postinit(
+        body=[],
+        args=inferred.args,
+        doc_node=getattr(inferred, "doc_node", None),
```

Comment:
```suggestion
        doc_node=inferred.doc_node
```
Both `FunctionDef` and `Lambda` should always have a `doc_node` attribute.

Reply:
This crashes currently as well. For some reason accessing `doc_node` or `doc` directly raises `StopIteration`. I felt like narrowing that issue down was out of scope. See also L573 on the old code. There is an pre-existing issue here.

Reply:
Just checked, `inferred` is narrowed to `FunctionDef | Lambda` and only `FunctionDef` has a `doc` / `doc_node` attribute. So `getattr` is correct here.